do not wrap reqwest::Client in Arc, as it itself uses Arc internally

utter-step opened the pull request:
Hello! Fist of all – thank you for your crate, I've been using it third time this week for my pet projects :)

I was reading `voyager` sources and noticed that you wrap `reqwest::Client` in an `Arc`. I've been doing it myself for some time, until I read thoroughly `reqwest` docs and noticed, that Client is [already][1] Arc-wrapped internaly.

If you are OK with it – in this PR I remove Arc-wrapping from voyager. 
There is also some questions about possibilities of breaking changes that I'd like to discuss, if you decide to merge this PR.

Thanks again,
Vlad :)

[1]: https://github.com/seanmonstar/reqwest/blob/master/src/async_impl/client.rs#L56-L63

utter-step commented on src/lib.rs:
```diff
@@ -367,11 +364,11 @@ where
     /// the scraper again
     pub fn crawl<TCrawlFunction, TCrawlFuture>(&mut self, fun: TCrawlFunction)
     where
-        TCrawlFunction: FnOnce(Arc<reqwest::Client>) -> TCrawlFuture,
```
this (and also `complete`) looks like internal Crawler function, but it is `pub`, so formally it is breaking change.

What's better – make this change (and up crate minor version, I guess?) or leave it as it is (almost, it'll use Arc::new instead of Arc::clone)?

mattsse commented on src/lib.rs:
```diff
@@ -609,12 +606,14 @@ impl CrawlerConfig {
-        self.client = Some(client);
+    /// *NOTE* [`reqwest::Client`] already uses Arc under the hood, so
+    /// it's preferable to just `clone` it and pass via [`Self::set_client`]
+    pub fn with_shared_client(mut self, client: std::sync::Arc<reqwest::Client>) -> Self {
```
```suggestion
    #[deprecated( since = "0.2", message = "You do not have to wrap the Client it in a `Arc` to reuse it, because it already uses an `Arc` internally. Users should instead use `set_client`instead."]
    pub fn with_shared_client(mut self, client: std::sync::Arc<reqwest::Client>) -> Self {
```

mattsse reviewed (approved):
Thanks for this,
old habits die hard.

I'm fine with this PR.
And if you are up for it, I'm happy to include another one that bumps the minor version.

utter-step commented:
Thanks for quick reaction!
I've updated version and added deprecation note, looks like your approval is needed once more :)

<img width="562" alt="Screenshot 2021-09-19 at 12 19 16" src="https://user-images.githubusercontent.com/829944/133922173-735055a0-cc01-4089-9556-2a3230d33f49.png">
